WooCommerce/Classes/ViewRelated/Products/SKU Scanner/CodeScannerViewController.swift

```diff
@@ -188,14 +207,30 @@ private extension BarcodeScannerViewController {
             guard self.session.isRunning, barcodes.isNotEmpty else {
                 return
             }
-            self.onBarcodeScanned(.success(barcodes))
+            completion(.success(barcodes))
+        }
+    }
+
+    func handleTextDetectionResults(request: VNRequest, error: Error?, completion: @escaping (Result<[String], Error>) -> Void) {
+        guard let textObservations = request.results?.compactMap({ $0 as? VNRecognizedTextObservation }) else {
+            return
+        }
+
+        let recognizedStrings = textObservations.compactMap { observation in
+            // Returns the string of the top `VNRecognizedText` instance.
+            observation.topCandidates(1).first?.string
+        }.uniqued()
+
+        DispatchQueue.main.async { [weak self] in
```

Comment:
[Q] Why does this needs to be dispatched in the next run loop?

Reply:
Good question - I was following the equivalent barcode implementation for this one, I believe this check on the next run loop is to ensure the camera session is still running while the scan results are ready because the detection & handling happen async.